docs/docs/themes/api-reference.md

```diff
@@ -56,39 +55,86 @@ module.exports = themeOptions => {
+### Extending
+
+In the last section we talked about completely replacing one component with another. What if we want to make a smaller change that doesn't require copy/pasting the entire theme component into our own? We can take advantage of the ability to extend components.
+
+Taking the `Header` example from before, when we write our shadowing file at `src/gatsby-theme-amazing/components/header.js`, we can import the original component and re-export it as such, adding our own overridden prop to the component.
 
 ```js
-import Layout from "gatsby-theme-tomato/src/components/Layout"
+import Header from "gatsby-theme-amazing/src/components/header"
+
+// these props are the same as the original component would get
+export default props => <Header {...props} myProp="true" />
 ```
 
-Gatsby Themes also allow you to customize any file in a theme's `src` directory by following a file naming convention.
-If you're using `gatsby-theme-tomato` which uses a `ProfileCard` component located at `src/components/profile-card.js` you can override the component by creating `src/gatsby-theme-tomato/components/profile-card.js`. If you want to see what props are passed you can do so by putting the props into a `pre` tag:
+Taking this approach means that when we upgrade our theme later we can also take advantage of all the updates to the `Header` component because we haven't fully replaced it, just modified it.
+
+### What path do use to shadow a file?
 
-```js:title=src/gatsby-theme-tomato/components/profile-card.js
-import React from "react"
+Until we build tooling to support automatically handling shadowing you will have to figure out how to convert paths in a theme into paths in your site on your own. Luckily the way to do that is few steps. Take the `src` directory from the theme, and move it to the front of the path, then write a file at that location in your site. Looking back on our `Header` example, this is the path to the component in our theme:
 
-export default props => <pre>{JSON.stringify(props, null, 2)}</pre>
 ```
+gatsby-theme-amazing/src/components/header.js
+```
+
+and here is the path where we would shadow it in our site:
+
+```
+<our-site>/src/gatsby-theme-amazing/components/header.js
+```
+
+Shadowing only works on imported files in the `src` directory. This is because shadowing is built on top of Webpack, so the module graph needs to include the shadowable file.
+
+Since we can use many themes in a given site, there are many potential places to shadow a given file (one for each theme and one for the user's site). In the event that there exist many files attempting to shadow `gatsby-theme-amazing/src/components/header.js`, the last one included in the plugins array will win although the user's site will beat any plugin.
```

Comment:
```suggestion
Since we can use multiple themes in a given site, there are many potential places to shadow a given file (one for each theme and one for the user's site). In the event that more than one themes are attempting to shadow `gatsby-theme-amazing/src/components/header.js`, the last theme included in the plugins array will win. The site itself takes the highest priority in shadowing.
```